src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,35 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private Map<String, UserData> cache;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
+        cache = new ConcurrentHashMap<>();
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.completedFuture(cache.get(userId));
+        }
+
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
```

Comment:
Catching InterruptedException is appropriate, but it would be helpful to log the exception as well to aid in debugging in case the interruption is unexpected.